Makefile-libostree.am

```diff
@@ -261,6 +261,10 @@ OSTree-1.0.gir: libostree-1.la Makefile
 OSTree_1_0_gir_EXPORT_PACKAGES = ostree-1
 OSTree_1_0_gir_INCLUDES = Gio-2.0
 OSTree_1_0_gir_CFLAGS = $(libostree_1_la_CFLAGS)
+if ENABLE_EXPERIMENTAL_API
+# When compiling this is set via config.h, but g-ir-scanner can't use that
+OSTree_1_0_gir_CFLAGS += -DOSTREE_ENABLE_EXPERIMENTAL_API=1
```

Comment:
I can’t remember where I commented about this before, but I think it’s possible to just pass `config.h` to `g-ir-scanner` in its list of files to scan (as the first file), and then it will end up being `#include`d in the generated C file which it compiles to introspect the GTypes.

I haven’t actually tried that though. Does it work?

Reply:
Yes, I did comment on this recently: https://github.com/ostreedev/ostree/pull/1292/files#r147126774

Did you try it and it didn’t work? :-(

Reply:
Sorry, meant to mention that. It's kind of in the commit message. I did try it and `g-ir-scanner` was not happy. It expects that all the macros are going to follow your namespace (i.e., `OSTREE_*`), but there are a ton of random macros in there like `LT_OBJDIR`. So, I think you can do it, but it will output a ton of warnings. I think for this one special case it's easier to just pass the flag on the command line.

Reply:
Cool, thanks for trying it. Could you [file a bug against gobject-introspection](https://bugzilla.gnome.org/enter_bug.cgi?product=gobject-introspection) about making this a bit nicer to use please? It’s a fairly common idiom, I think, and it would be nice if `g-ir-scanner` made it a bit easier to pass in `config.h`.

Reply:
Done - https://bugzilla.gnome.org/show_bug.cgi?id=789881. I honestly don't know if much can be done, though, since I don't know how g-ir-scanner would know the difference between random macros you want to ignore and macros you want parsed that don't follow your namespace.